src/core/packages/saved-objects/base-server-internal/src/model_version/version_map.ts

```diff
@@ -64,12 +64,11 @@ export const getModelVersionMapForTypes = (types: SavedObjectsType[]): ModelVers
 
 /**
  * Returns the current virtual version for the given type.
- * It will either be the latest model version if the type
- * already switched to using them (switchToModelVersionAt is set),
- * or the latest migration version for the type otherwise.
+ * It will either be the latest model version or the latest
+ * migration version for the type if model versions have not been declared.
  */
 export const getCurrentVirtualVersion = (type: SavedObjectsType): string => {
-  if (type.switchToModelVersionAt) {
+  if (type.modelVersions) {
```

Comment:
One of the main changes in this PR is to assume we can relax the enforced switch to modelVersions. There is a risk that type owners continue to add migrations but with the pending removal of migrations from the SO type, we hope that this doesn't happen.